### ISSUE TITLE
Incorrect serializetion of SplFixedArray

### ISSUE THREAD

**Issue by vakarelov** [facebook/hhvm#7513]
### HHVM Version
3.15.3
### Standalone code, or other way to reproduce the problem

$test_fixed_array = New SplFixedArray(3);
$test_fixed_array[0] = 0;
$test_fixed_array[1] = 1;
$test_fixed_array[2] = 2;

$ser = serialize($test_fixed_array);

$test_fixed_array_recover = unserialize($ser);

var_dump($test_fixed_array_recover);

file_put_contents("test.txt", $ser);

### Expected result
From PHP 7
The file:
O:13:"SplFixedArray":3:{i:0;i:0;i:1;i:1;i:2;i:2;}

Dump:
object(SplFixedArray)[3]
  public 0 => int 0
  public 1 => int 1
  public 2 => int 2

### Actual result
The file:
O:13:"SplFixedArray":1:{s:7:" * data";a:3:{i:0;i:0;i:1;i:1;i:2;i:2;}}

Dump:
trueobject(SplFixedArray)#4 (1) { ["data":protected]=> array(3) { [0]=> int(0) [1]=> int(1) [2]=> int(2) } } 

Basically, the SplFixedArray becomes a one-element array containing an ordinary three-element array

**Comment by azjezz** [comment]
should be closed.

php is not supported any more, therefore incompatibility with php is not considered an issue.
